cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -81,8 +81,20 @@ macro(swift_common_standalone_build_config_llvm product is_cross_compiling)
   # HACK: this ugly tweaking is to prevent the propagation of the flag from LLVM
   # into swift.  The use of this flag pollutes all targets, and we are not able
   # to remove it on a per-target basis which breaks cross-compilation.
+  # Therefore, we always remove it here and then selectively re-add it to the
```

Comment:
I have noticed this hack making the rounds recently. What is going on here? Why doesn't LLVM run into these issues as well?

Reply:
The short version is Windows doesn't support `-z,defs` as a linker flag, and `CMAKE_SHARED_LINKER_FLAGS` get applied to all linker invocations, so we have to move where it gets added.

LLVM doesn't run into it because it uses the `CMAKE_SYSTEM_NAME` method for cross-compiling (so this flag would never get inserted in the first place for Windows), whereas Swift has a custom scheme where we don't know the SDK targets until much later in the configuration process.

Reply:
This explanation should be added here. Also, it should be clear that this is due to windows.